patches/server/1049-Propagate-failed-to-bind-to-port-error.patch

```diff
@@ -0,0 +1,18 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Warrior <50800980+Warriorrrr@users.noreply.github.com>
+Date: Sat, 10 Feb 2024 10:28:50 +0100
+Subject: [PATCH] Propagate failed to bind to port error
+
+
+diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+index 6de6dad36203479677a29ad61e21bc369d4e5513..2ecf79086e9e8069f43120e9146e98962a75eca8 100644
+--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
++++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+@@ -265,6 +265,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
+             DedicatedServer.LOGGER.warn("**** FAILED TO BIND TO PORT!");
+             DedicatedServer.LOGGER.warn("The exception was: {}", ioexception.toString());
+             DedicatedServer.LOGGER.warn("Perhaps a server is already running on that port?");
++            if (true) throw new IllegalStateException("Failed to bind to port", ioexception); // Paper - Propagate failed to bind to port error
```

Comment:
Just out of curiosity, why do you check `if (true)` here?

Reply:
To avoid it from complaining about the return being unreachable, I thought it'd be better to do it like so to make it only 1 line of diff but it probably doesn't matter much

Reply:
Ohh yeah that makes sense. Thanks for clarification :)